fix(bazel): pin terser to 4.4.0

alan-agius4 opened the pull request:
4.4.1 caused a regression https://github.com/terser/terser/issues/527 which is breaking our CI with such errors
```
'http://localhost:4200/main.57ce335e666ac98d65ad.js 0:370275 TypeError: e is not a function
```

Example of failing build https://circleci.com/gh/angular/angular/549169?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

gkalpak commented on packages/bazel/src/schematics/ng-add/index.ts:
```diff
@@ -59,7 +59,7 @@ function addDevDependenciesToPackageJson(options: Schema) {
       'rollup-plugin-commonjs': '^10.1.0',
       'rollup-plugin-node-resolve': '^5.2.0',
-      'terser': '^4.3.9',
+      'terser': '4.4.0',
```
Just to make sure I understand correctly, is this only needed in case we publish a `@angular/bazel` version before terser 4.4.2 is released, right?

Once 4.4.2 is released, should this be changed to `^4.4.2` again?

alan-agius4 replied:
Yes

gkalpak commented on integration/bazel-schematics/package.json:
```diff
@@ -8,6 +8,9 @@
     "@bazel/bazel": "file:../../node_modules/@bazel/bazel"
   },
+  "resolutions": {
+    "terser": "4.4.0"
```
Could you add a comment here (à la https://github.com/angular/angular/commit/ad753d3fa7f5d45cc72c9d90b2aeff7eb437f779#diff-0f512cda5dff463a54c9e4db5cc1ad3cR178-R180), pointing to the terser issue and mentioning when this can be removed?

alan-agius4 replied:
Yeah, I am still looking at it. The problem is that in this test we are hitting hoisting issues.

alan-agius4 replied:
Should be fixed now

alan-agius4 committed:
fix(bazel): pin terser to 4.4.0

4.4.1 caused a regression https://github.com/terser/terser/issues/527 which is breaking our CI with such errors:
```
http://localhost:4200/main.57ce335e666ac98d65ad.js 0:370275 TypeError: e is not a function
```

Example of failing build https://circleci.com/gh/angular/angular/549169?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

josephperrott reviewed (approved):
LGTM

alan-agius4 commented:
It seems that terser fixed the bug https://github.com/terser/terser/issues/527

Thus, I am going to close this as it's no longer needed.
